blaze-server/src/main/scala/org/http4s/server/blaze/Http1ServerStage.scala

```diff
@@ -253,12 +257,21 @@ private[blaze] class Http1ServerStage[F[_]](
   override protected def stageShutdown(): Unit = {
     logger.debug("Shutting down HttpPipeline")
     parser.synchronized {
+      cancel()
```

Comment:
Does this signal cancelled even on correct shutdown?

Reply:
I think it would.  We could blank it out, but I had to synchronize on the parser to make it reliable, and we'd have to think through the deadlock ramifications.

Alternatively: make the test pass without synchronization.

Reply:
Detecting correct shutdown is going to be tricky here. Maybe it's more appropriate to cancel only when we receive a `Disconnected` message?